src/librustdoc/html/render/mod.rs

```diff
@@ -3517,14 +3520,19 @@ fn render_assoc_items(
                 RenderMode::Normal
             }
             AssocItemRender::DerefFor { trait_, type_, deref_mut_ } => {
+                let id =
+                    cx.derive_id(small_url_encode(&format!("deref-methods-{:#}", type_.print())));
+                cx.deref_id_map.borrow_mut().insert(type_.def_id().unwrap(), id.clone());
                 write!(
                     w,
-                    "<h2 id=\"deref-methods\" class=\"small-section-header\">\
+                    "<h2 id=\"{}\" class=\"small-section-header\">\
                          Methods from {}&lt;Target = {}&gt;\
-                         <a href=\"#deref-methods\" class=\"anchor\"></a>\
+                         <a href=\"#{}\" class=\"anchor\"></a>\
                      </h2>",
+                    id,
                     trait_.print(),
-                    type_.print()
+                    type_.print(),
+                    id,
```

Comment:
You can have named arguments, no need to pass `id` twice. ;)

Reply:
Ah indeed thanks, I'll clean this up. 😄